flake.nix

```diff
@@ -0,0 +1,124 @@
+{
+  description = "Build teos (The Eyes of Satoshi) server and plugin";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+    crane = {
+      url = "github:ipetkov/crane";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+
+    fenix = {
+      url = "github:nix-community/fenix";
+      inputs.nixpkgs.follows = "nixpkgs";
+      inputs.rust-analyzer-src.follows = "";
+    };
+
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, crane, fenix, flake-utils, ... }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+
+        inherit (pkgs) lib;
+
+        craneLib = crane.mkLib pkgs;
+        src = craneLib.cleanCargoSource ./.;
+
+        # Common arguments can be set here to avoid repeating them later
+        commonArgs = {
+          inherit src;
+          strictDeps = true;
+ 
+	  nativeBuildInputs = [ 
+            pkgs.pkg-config 
+            pkgs.openssl 
+            pkgs.rustfmt # needed for tonic build
+          ];
+
+          buildInputs = [
+            # Add additional build inputs here
+            pkgs.pkg-config
+          ] ++ lib.optionals pkgs.stdenv.isDarwin [
+            # Additional darwin specific inputs can be set here
+            pkgs.libiconv
+          ];
+
+          # TODO: hack but without there are warnings and I can't make other method work 
```

Comment:
I'm curious about what the issue with this is. I'm not familiar with Nix so just wondering

Reply:
The issue is that changing the released version in Cargo.toml require changing the version here, while instead the value should be read from the `Cargo.toml` thus not requiring to keep this file in sync.

Reply:
Crane is expecting a version in the workspace Cargo.toml. So either you can add one or it won't be possible to build just the dependencies (without warnings).